src/components/Spawn/editHostModal/useLoadFormData.ts

```diff
@@ -0,0 +1,58 @@
+import { useQuery } from "@apollo/client";
+import {
+  GetMyPublicKeysQuery,
+  GetMyPublicKeysQueryVariables,
+  InstanceTypesQuery,
+  InstanceTypesQueryVariables,
+  MyVolumesQuery,
+  MyVolumesQueryVariables,
+} from "gql/generated/types";
+import {
+  GET_INSTANCE_TYPES,
+  GET_MY_PUBLIC_KEYS,
+  GET_MY_VOLUMES,
+} from "gql/queries";
+import { useDisableSpawnExpirationCheckbox, useSpruceConfig } from "hooks";
+import { MyHost } from "types/spawn";
+import { getNoExpirationCheckboxTooltipCopy } from "../utils";
+
+export const useLoadFormData = (host: MyHost) => {
```

Comment:
Great idea with the hook!

Reply:
all credit goes to arjun who did it before me :P